tangelo/linq/tests/test_simulator.py

```diff
@@ -368,6 +368,37 @@ def test_get_exp_value_from_statevector_h2(self):
         if test_fail:
             assert False
 
+    def test_get_exp_value_h2_with_n_shots(self):
+        """ Get expectation value of large circuits and qubit Hamiltonians corresponding to molecules.
+            Molecule: H2 sto-3g = [("H", (0., 0., 0.)), ("H", (0., 0., 0.741377))] with increasing shots
+            obtains better accuracy but never the exact result.
+        """
+        qubit_operator = load_operator("mol_H2_qubitham.data", data_directory=path_data, plain_text=True)
+
+        with open(f"{path_data}/H2_UCCSD.qasm", "r") as circ_handle:
+            openqasm_circ = circ_handle.read()
+
+        abs_circ = translate_c(openqasm_circ, "tangelo", source="openqasm")
+        expected = -1.1372704
+        test_fail = False
+
+        for n_shots in [10**2, 10**4, 10**6]:
+            for b in installed_sv_simulator:
+                sim = get_backend(target=b, n_shots=n_shots)
+                tstart = time.time()
+                energy = sim.get_expectation_value(qubit_operator, abs_circ)
+                tstop = time.time()
+                print(f"H2 get exp value with {b:10s} with n_shots={n_shots} returned {energy:.7f} \t Elapsed: {tstop - tstart:.3f} s.")
+
+                try:
+                    self.assertNotAlmostEqual(energy, expected, delta=1.e-6)
+                    self.assertAlmostEqual(energy, expected, delta=10/np.sqrt(n_shots))
+                except AssertionError:
+                    test_fail = True
+                    print(f"{self._testMethodName} : Assertion failed {b} (result = {energy:.7f}, expected = {expected})")
+            if test_fail:
```

Comment:
`assert not test_fail` is better.

I assume at this point every single backend who failed returned a print, but we allowed everything to continue to get all the info, and only that assert decides whether or not the overall test in a success.